### PR TITLE
Update ERC721 token

### PR DESCRIPTION
**Pull request by nrryuya**
### - What I did
Update example implementation of [ERC721](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md) token. Specifically, 
- supported ERC165 (it's required in the EIP)
- added mint/burn function
- made it clear that assertions required in the EIP is implemented. 
- commented about other assertions not included in the EIP but needed.
- added tests with more test cases

### - How I did it
I referred to the previous implementation and [OpenZeppelin's Solidity implementation](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/token/ERC721/ERC721.sol).

### - How to verify it
Test. Formal verification is [WIP](https://github.com/LayerXcom/verified-vyper-contracts/tree/master/erc721). 

### - Description for the changelog
Update example ERC721 token

### - Cute Animal Picture
image

### CODE REVIEW

File: examples/tokens/ERC721.vy [hunk 1]
```diff
@@ -62,18 +62,39 @@ ownerToNFTokenCount: uint256[address]
 # @dev Mapping from owner address to mapping of operator addresses.
 ownerToOperators: (bool[address])[address]
 
+# @dev Address of minter, who can mint a token
+minter: address
 
-# @dev Contract constructor. Per the Transfer event spec; during contract creation, any number of
-#      NFTs may be created and assigned without emitting Transfer.
+# @dev Mapping of interface id to bool about whether or not it's supported
+supportedInterfaces: bool[bytes32]
```
Comment (fubuloubu) [inline examples/tokens/ERC721.vy]: Does `public` generate the correct getter `supportsInterface(_interfaceID: bytes32) -> bool`?

Comment (fubuloubu) [reply examples/tokens/ERC721.vy]: hmm, yeah I guess "explicit is better than implicit", just seems like a waste of `public`'s utility

**Comment by nrryuya** [comment]
Added a few minor fixes

File: examples/tokens/ERC721.vy (hunk 1 above)
Comment (nrryuya) [reply examples/tokens/ERC721.vy]: >seems like a waste of public's utility

Yeah that's also true.

**Comment by fulldecent** [comment]
Please compare also to the reference implementation at https://github.com/0xcert/ethereum-erc721

**Pull request merged**